spec/models/reservation_spec.rb

```diff
@@ -87,705 +73,830 @@
+      expect(res.approved?).to be_falsey
+    end
+    it 'returns true if approved' do
+      res = FactoryGirl.build_stubbed(:request, status: 'reserved')
+      expect(res.approved?).to be_truthy
+    end
+    it 'returns false if not a request' do
+      res = FactoryGirl.build_stubbed(:valid_reservation)
+      expect(res.approved?).to be_falsey
     end
   end
 
-  context 'with blacked out start date' do
-    let!(:blackout) do
-      FactoryGirl.create(:blackout,
-                         start_date: reservation.start_date,
-                         end_date: reservation.due_date)
+  describe '#flagged?' do
+    let!(:res) { FactoryGirl.build_stubbed(:valid_reservation) }
+    it 'returns true when flagged' do
+      res.flag(:request)
+      expect(res.flagged?(:request)).to be_truthy
     end
-
-    it { is_expected.to be_valid }
-    it 'should save' do
-      expect(reservation.save).to be_truthy
-      expect(Reservation.all.size).to eq(1)
-    end
-    it 'can be updated' do
-      reservation.start_date = Time.zone.today + 1.day
-      expect(reservation.save).to be_truthy
-    end
-    it 'fails appropriate validations' do
-      expect(reservation.validate).not_to eq([])
-    end
-    it 'passes other custom validations' do
-      expect(reservation.not_in_past).to be_nil
-      expect(reservation.start_date_before_due_date).to be_nil
-      expect(reservation.not_empty).to be_nil
-      expect(reservation.matched_item_and_model).to be_nil
-      expect(reservation.available).to be_nil
-      expect(reservation.check_banned).to be_nil
+    it 'returns false when not flagged' do
+      expect(res.flagged?(:request)).to be_falsey
+    end
+    it 'returns false when the flag is undefined' do
+      expect(res.flagged?(:garbage_flag)).to be_falsey
     end
   end
 
-  context 'with no user' do
-    subject(:reservation) do
-      FactoryGirl.build(:valid_reservation, reserver: nil)
+  describe '#attrs?' do
+    it 'returns true when all attributes match' do
+      attrs = { overdue: true, status: 'checked_out' }
+      res = FactoryGirl.build_stubbed(:overdue_reservation)
+      expect(res.attrs?(attrs)).to be_truthy
     end
-
-    it 'should have a deleted user' do
-      expect(reservation.reserver).not_to be_nil
-      expect(reservation.reserver.first_name).to eq('Deleted')
+    it 'returns false when one attribute does not match' do
+      attrs = { overdue: true, status: 'checked_out' }
+      res = FactoryGirl.build_stubbed(:checked_out_reservation)
+      expect(res.attrs?(attrs)).to be_falsey
     end
-    it { is_expected.to be_valid }
   end
 
-  context 'when user has overdue reservation' do
-    subject(:reservation) { FactoryGirl.build(:valid_reservation) }
-    let(:overdue_reserver) { reservation.reserver }
-    let!(:overdue) do
-      o = FactoryGirl.build(:overdue_reservation, reserver: overdue_reserver)
-      o.save(validate: false)
-      o
+  describe '#overlaps_with' do
```

Comment:
True! I think it should probably be used in many other places (we check overlap quite a bit), but it can be private until the need arises to use it.